cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -18,72 +18,205 @@ package options
-		// Please make common changes there and put anything cloud specific here.
-		Generic:                   cmoptions.NewGenericControllerManagerOptions(componentConfig),
-		NodeStatusUpdateFrequency: metav1.Duration{Duration: 5 * time.Minute},
+		CloudProvider: &cmoptions.CloudProviderOptions{},
+		Debugging:     &cmoptions.DebuggingOptions{},
+		GenericComponent: &cmoptions.GenericComponentConfigOptions{
+			MinResyncPeriod:         componentConfig.GenericComponent.MinResyncPeriod,
+			ContentType:             componentConfig.GenericComponent.ContentType,
+			KubeAPIQPS:              componentConfig.GenericComponent.KubeAPIQPS,
+			KubeAPIBurst:            componentConfig.GenericComponent.KubeAPIBurst,
+			ControllerStartInterval: componentConfig.GenericComponent.ControllerStartInterval,
+			LeaderElection:          componentConfig.GenericComponent.LeaderElection,
+		},
+		KubeCloudShared: &cmoptions.KubeCloudSharedOptions{
+			Port:                      componentConfig.KubeCloudShared.Port,
+			Address:                   componentConfig.KubeCloudShared.Address,
+			RouteReconciliationPeriod: componentConfig.KubeCloudShared.RouteReconciliationPeriod,
+			NodeMonitorPeriod:         componentConfig.KubeCloudShared.NodeMonitorPeriod,
+			ClusterName:               componentConfig.KubeCloudShared.ClusterName,
+			ConfigureCloudRoutes:      componentConfig.KubeCloudShared.ConfigureCloudRoutes,
+		},
+		ServiceController: &cmoptions.ServiceControllerOptions{
+			ConcurrentServiceSyncs: componentConfig.ServiceController.ConcurrentServiceSyncs,
+		},
+		SecureServing: apiserveroptions.NewSecureServingOptions(),
+		InsecureServing: &cmoptions.InsecureServingOptions{
+			BindAddress: net.ParseIP(componentConfig.KubeCloudShared.Address),
+			BindPort:    int(componentConfig.KubeCloudShared.Port),
+			BindNetwork: "tcp",
+		},
+		Authentication:            nil, // TODO: enable with apiserveroptions.NewDelegatingAuthenticationOptions()
+		Authorization:             nil, // TODO: enable with apiserveroptions.NewDelegatingAuthorizationOptions()
+		NodeStatusUpdateFrequency: componentConfig.NodeStatusUpdateFrequency,
 	}
 
-	s.Generic.SecureServing.ServerCert.CertDirectory = "/var/run/kubernetes"
-	s.Generic.SecureServing.ServerCert.PairName = "cloud-controller-manager"
+	s.SecureServing.ServerCert.CertDirectory = "/var/run/kubernetes"
+	s.SecureServing.ServerCert.PairName = "cloud-controller-manager"
+
+	// disable secure serving for now
+	// TODO: enable HTTPS by default
+	s.SecureServing.BindPort = 0
 
 	return &s
 }
 
+// NewDefaultComponentConfig returns cloud-controller manager configuration object.
+func NewDefaultComponentConfig(insecurePort int32) componentconfig.CloudControllerManagerConfiguration {
+	scheme := runtime.NewScheme()
+	componentconfigv1alpha1.AddToScheme(scheme)
+	componentconfig.AddToScheme(scheme)
+	versioned := componentconfigv1alpha1.CloudControllerManagerConfiguration{}
```

Comment:
add some empty line to make the code clearer.

Reply:
Maybe make scheme into a init func is a better choice like scheduler does.